docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,243 @@
+
+If you are unfamiliar with the functionality of PivotTables, consider exploring them as an end-user. 
+See [Create a PivotTable to analyze worksheet data](https://support.office.com/en-us/article/Import-and-analyze-data-ccd3c4a6-272f-4c97-afbb-d3f27407fcde#ID0EAABAAA=PivotTables) for a good primer on these tools. 
+
+This article provides code samples for common scenarios.
+For the complete list of properties and methods the **PivotTable** and **PivotTableCollection** objects support, see [PivotTable API](https://dev.office.com/reference/add-ins/excel/pivottable) and [PivotTable Collection API](https://dev.office.com/reference/add-ins/excel/pivottablecollection).
+
+> [!NOTE]
+> These samples use APIs currently available only in public preview (beta). To run these samples, you must use the beta library of the Office.js CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
+
+## Hierarchies
+
+PivotTables are organized based on four hierarchy categories: row, column, data, and filter. The following data describing fruit sales from various farms will be used in throughout this article.
+
+![A collection of fruit sales of different types from different farms.](../images/excel-pivots-raw-data.png)
+
+This data has five hierarchies: **Farms**, **Type**, **Classification**, **Crates Sold at Farm**, and **Crates Sold Wholesale**. Each hierarchy can only exist in one of the four categories. If **Type** is added to column hierarchies and then added to row hierarchies, it only remains in the latter.
+
+Row and column hierarchies define how data will be grouped. For example, having a row hierarchy of **Farms** groups together all data sets from the same farm. The choice between row and column hierarchy defines the orientation of the PivotTable.
+
+Data hierarchies are the values to be aggregated based on the row and column hierarchies. A PivotTable with a row hierarchy of **Farms** and a data hierarchy of **Crates Sold Wholesale** shows the sum total (by default) of all the different fruits for each farm.
+
+Filter hierarchies include or exclude data from the pivot based on values within that filtered type. A filter hierarchy of **Classification** with the type **Organic** selected only shows data for organic fruit.
+
+Here is the farm data again, alongside a PivotTable. The PivotTable is using **Farm** and **Type** as the row hierarchies, **Crates Sold at Farm** and **Crates Sold Wholesale** as the data hierarchies (with the default aggregation function of sum), and **Classification** as a filter hierarchy (with **Organic** selected). 
+
+![A selection of fruit sales data next to a PivotTable with row, data, and filter hierarchies.](../images/excel-pivot-table-and-data.png)
+
+This PivotTable could be generated through the JavaScript API or through the Excel UI. Both options allow for further manipulation through add-ins.
+
+## Create a PivotTable
```

Comment:
please add a reference to a scriptlab yaml so people can give it a try!

Reply:
it should include all the samples below. thanks

Reply:
also make sure it includes a setup option so the needed worksheets are avialable

Reply:
What is our policy on linking to ScriptLab samples? @Reezaali? 

I definitely want to show off that sample I wrote, but none of the other "Core Concept" pages link to those yaml files.